### PR TITLE
Invalid region tag, Intl.DisplayNames.prototype.of()

### PR DESCRIPTION
**Pull request by ben-allen**
Added missing test for invalid "region" type in Intl.DisplayNames.of()

### CODE REVIEW

File: test/intl402/DisplayNames/prototype/of/type-region-invalid.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Igalia S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-Intl.DisplayNames.prototype.of
+description: Throws a RangeError for invalid `region` codes
+info: |
+  12.3.3 Intl.DisplayNames.prototype.of ( code )
```
Comment (Ms2ger) [inline test/intl402/DisplayNames/prototype/of/type-region-invalid.js:7]:
```suggestion
  12.5.1 CanonicalCodeForDisplayNames ( type, code )
```

File: test/intl402/DisplayNames/prototype/of/type-region-invalid.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Igalia S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-Intl.DisplayNames.prototype.of
+description: Throws a RangeError for invalid `region` codes
+info: |
```
Comment (Ms2ger) [inline test/intl402/DisplayNames/prototype/of/type-region-invalid.js:6]: Is https://unicode.org/reports/tr35/#unicode-language-identifier the relevant definition? If so, please add the link and copy the EBNF in here as well

**Review by Ms2ger — APPROVED** [review]

**Comment by ptomato** [comment]
@ben-allen This is probably ready to land when Ms2ger's remaining comment is addressed.

File: test/intl402/DisplayNames/prototype/of/type-region-invalid.js (hunk 2 above)
Comment (ben-allen) [reply test/intl402/DisplayNames/prototype/of/type-region-invalid.js:6]: updated!

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**